client/src/main/scala/com/redi/scalademo/presentation/FormListener.scala

```diff
@@ -95,34 +75,66 @@ class FormListener(
           formElement.find(elementNamed(name)).addClass(DangerClass)
         }
       }
-      name match {
-        case Augend ⇒
-          augend = value
-        case Addend ⇒
-          addend = value
+    }
+    allFormValuesAreValid
+  }
+
+  private def serializedFormData: js.Array[js.Dictionary[String]] = {
+    formElement.serializeArray().asInstanceOf[js.Array[js.Dictionary[String]]]
+  }
+
+  private def calculableFormData: js.Array[js.Dictionary[String]] = {
+    var calculableFormData = js.Array[js.Dictionary[String]]()
+    for (formControlData: js.Dictionary[String] ← serializedFormData) {
+      if (formControlData(Name) != Validate) {
+        calculableFormData += formControlData
+      }
+    }
+    calculableFormData
+  }
+
+  private def shouldValidate: Boolean = {
+    var shouldValidate = false
+    for (formControlData: js.Dictionary[String] ← serializedFormData) {
+      if (formControlData(Name) == Validate) {
+        shouldValidate &= formControlData(Value).nonEmpty
```

Comment:
I'm on the fence whether we should use `&=` over `=` since it will never actually do anything different. There's only one element that will hit this code path.